### PR TITLE
Chat activities

### PR DESCRIPTION
**Pull request by patrixr**
# Activities

Activities, as defined by Sunshine, are the following events:
- A user starts typing
- A user stops typing
- A user has read the conversation

This PR implements GraphQL mutations to alert the customer (on FB/Whatsapp/Telegram) of those events.

The mutations are the following:

- `startTyping(conversationId: Int!)`
- `stopTyping(conversationId: Int!)`
- `marAsRead(conversationId: Int!)`

### CODE REVIEW

File: lib/services/abilities/abilities.ts [hunk 1]
```diff
@@ -0,0 +1,218 @@
+import { AuthorType, ConversationType, Message, Staff }     from "@prisma/client"
```
Comment (patrixr) [inline lib/services/abilities/abilities.ts:1]: NOTE: this file was renamed and moved.

No new code was added here. You can ignore

File: prisma/schema.prisma [hunk 2]
```diff
@@ -17,7 +17,6 @@ model Staff {
   externalId          String   @db.VarChar(255) @unique
```
Comment (patrixr) [inline prisma/schema.prisma:17]: Note: Look here for database changes

File: lib/services/abilities/activities.ts [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+import { throwForbidden }                                from "../../utils/errors"
+import { ActivitiesApi }                                 from "sunshine-conversations-client"
+import { abilities }                                     from "./abilities"
+import config                                            from "../../config"
+import db                                                from "../../db"
+
+export function activities(staff: Staff) {
+
+  const crud = abilities(staff);
+  const sunchineActivities = new ActivitiesApi();
```
Comment (steveyken) [inline lib/services/abilities/activities.ts]:
```suggestion
  const sunshineActivities = new ActivitiesApi();
```

File: lib/services/abilities/activities.ts [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+
+  // --- Helpers
+
+  const triggerSunshineActivity = (sunshineConversationId: string, type: string) => {
+    return sunchineActivities.postActivity(config.smoochAppId, sunshineConversationId, {
```
Comment (steveyken) [inline lib/services/abilities/activities.ts]:
```suggestion
    return sunshineActivities.postActivity(config.smoochAppId, sunshineConversationId, {
```

**Review by steveyken — APPROVED** [review]
💯 just a small typo!

File: lib/services/abilities/activities.ts (hunk 3 above)
Comment (patrixr) [reply lib/services/abilities/activities.ts]: thanks

**Pull request merged**